### ISSUE TITLE
add road information to the elevation profile

### ISSUE THREAD

**Issue by basings** [osmandapp/OsmAnd#14100]
# 🚀 feature request

### Description

Currently, you can select to colorize the route basedon the surface of the road. It would be great to have the information below the elevation profile as a bar that's colorized based on the road type and/or based on the surface. It would also be great to have a legend to see what each color represents. This can help in deciding if you want to take the route, e.g. for a bike ride if you want to drive over longer unpaved segments (without having to disable unpaved roads) or if you want to drive over a bigger road. Even though you are allowed to, but you may not want to drive over such roads.

### Describe the solution you'd like

A bar below the elevation profile. It could be hidden below the screen and you could draw the hole elevation profile popup up in order to see it. Or you could cycle through the different graphs (elevation profile/ road type) (with a button left/right).

### Describe alternatives you've considered

Only colorizing the path on the screen but that's not as useful as a separate graphics.

**Comment by tradavyvy** [comment]
👍👍

**Comment by njmdietrich** [comment]
This information is already available when using osmand's offline routing engine - just press "Details" on the navigation prompt. I would, however, also like this feature for "generic" gpx tracks and when using other routing engines such as brouter.

**Comment by basings** [comment]
@njmdietrich Wow! Thank you! I seriously should read an osmand features book. In that case, this is superfluous. ShouldIopen a new issue for the use case of gpx tracks?

**Comment by Gbernardy** [comment]
Yes it's true, the information Road type, surface, smoothness, steepness and surface firmness are already available but cannot be analyzed on the path/route as it is already possible for altitude, slope and speed.

A solution could be to add the relative items in the following menu.

![IMG_20220422_124726](https://user-images.githubusercontent.com/99968793/164700568-f5b11d79-b659-43bf-a3b2-0a41bca34b2d.jpg)

![IMG_20220422_125141](https://user-images.githubusercontent.com/99968793/164701118-dbb8cf89-36f6-40ac-ad1c-3d14de223189.jpg)

With this solution, the information would also become available on the gpx tracks, as long as this information is already contained in the same tracks.

**Comment by vshcherb** [comment]
It looks like we are talking about extra information for Elevation Widget. Did I understand correctly?

![telegram-cloud-photo-size-2-5407083618158885526-x](https://user-images.githubusercontent.com/1042025/164725759-c6c5d49d-27b7-46f6-a143-be69afafcdbd.jpg)

**Comment by basings** [comment]
@Gbernardy problem is probably that road information is not contained within a gpx track and needs to be fetched seperately, right? 

Good idea to put it into that menu!

**Comment by basings** [comment]
@vshcherb Yes, exactly!

**Comment by njmdietrich** [comment]
> @Gbernardy problem is probably that road information is not contained within a gpx track and needs to be fetched seperately, right?
> 
> Good idea to put it into that menu!

I have found a way to get this information for gpx tracks as well, it is unfortunately somewhat convoluted. It requires you to let osmand navigate you along the gpx track, but snapping to roads using a navigation profile of your choosing. Then you just have to make it start at the start of the route, not your current position, and there it is.

**Comment by Gbernardy** [comment]
Yes, but in any case it is not possible to analyze / compare the road type, surface, ... information on the map, as it is possible for altitude, slope and speed.  

When you click on analyze on map, that information cannot be analyzed anyway.

![Screenshot_2022-04-23-08-21-01-66_92b64b2a7aa6eb3771ed6e18d0029815](https://user-images.githubusercontent.com/99968793/164882997-d2d046fc-3985-4f37-8eb7-a2653fff08f1.jpg)
